bin/ember-template-lint.js

```diff
@@ -274,20 +293,39 @@ async function run() {
       filePaths.has(STDIN)
     );
 
-    let messages = await lintSource(linter, linterOptions, options.fix);
+    let fileResults;
 
-    resultsAccumulator.push(...messages);
+    if (options.fix) {
+      let { isFixed, output, messages } = await linter.verifyAndFix(linterOptions);
+      if (isFixed) {
+        fs.writeFileSync(linterOptions.filePath, output);
+      }
+      fileResults = messages;
+    } else {
+      fileResults = await linter.verify(linterOptions);
+    }
+
+    if (options.updateTodo) {
+      await linter.updateTodo(linterOptions, fileResults);
+    }
+
+    if (!filePaths.has(STDIN)) {
+      fileResults = await linter.processTodos(linterOptions, fileResults);
+    }
+
+    resultsAccumulator.push(...fileResults);
   }
 
   let results = processResults(resultsAccumulator);
+
   if (results.errorCount > 0) {
     process.exitCode = 1;
   }
 
   if (options.printPending) {
     return printPending(results, options);
   } else {
-    if (results.errorCount || results.warningCount) {
+    if (results.errorCount || results.warningCount || (results.todoCount && options.includeTodo)) {
```

Comment:
I know it's not a huge thing, but logically it makes sense to reverse this condition, since we want to primarily check if `includeTodo` is `true`, _then_ check if there's a count. Logically, we should be short circuiting if `includeTodo` is `false`.

```suggestion
    if (results.errorCount || results.warningCount || (options.includeTodo || results.todoCount)) {
```

Reply:
I suppose it won't short circuit anyway, since it's an or...